jobs/v4/howto/delete_company_sample.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package howto
+
+// [START job_search_delete_company]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	talent "cloud.google.com/go/talent/apiv4beta1"
+	talentpb "google.golang.org/genproto/googleapis/cloud/talent/v4beta1"
+)
+
+// deleteCompany deletes an existing company by name. Companies with
+// existing jobs cannot be deleted until those jobs have been deleted.
+func deleteCompany(w io.Writer, companyName string) error {
```

Comment:
There isn't a method to get `companyID` for [`type: Company`](https://godoc.org/google.golang.org/genproto/googleapis/cloud/talent/v4beta1#Company). Users would have to deconstruct `companyName` then reconstruct it. Is it necessary?

Reply:
See existing Go sample: https://github.com/GoogleCloudPlatform/golang-samples/blob/master/vision/product_search/get_similar_products.go#L30-L52

Excerpt:

```go
func getSimilarProducts(w io.Writer, projectID string, location string, productSetID string, productCategory string, file string, filter string) error {
	// ...
	ictx := &visionpb.ImageContext{
		ProductSearchParams: &visionpb.ProductSearchParams{
			ProductSet:        fmt.Sprintf("projects/%s/locations/%s/productSets/%s", projectID, location, productSetID),
			ProductCategories: []string{productCategory},
			Filter:            filter,
		},
	}
```

Because Go does not offer helper functions for path construction (the other 6 languages do), it is very useful to show `field: fmt.Sprint("projects/%s/companies/%s", projectId, companyId)`. Showing `field: companyName` doesn't help the user reading this sample figure out how to call it correctly.

If I, personally, saw a function accepting `companyName`, I would probably try to call it using the Display Name I used when creating the company. The fact that all GCP API resource objects have a string name field in the format `projects/x/...` is not immediately obvious to people.